src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -181,10 +181,7 @@ private protected override TPredictor TrainModelCore(TrainContext context)
             {
                 var initPred = context.InitialPredictor;
                 var linearInitPred = initPred as LinearModelParameters;
-                if (linearInitPred == null)
-                    linearInitPred = ((initPred as IWeaklyTypedCalibratedModelParameters)?.WeeklyTypedSubModel) as LinearModelParameters;
-
-                // If initial predictor is set, it must be a linear model or calibrated linear model. Otherwise, we throw.
+                // If initial predictor is set, it must be a linear model.
```

Comment:
We might want to allow two model types.

- LinearModelParameters
- CalibratedModelParametersBase<BinaryLinearModelParameters, PlattCalibrator>
The change below seems removing the second case.

Reply:
I'd rather not, honestly, since it's just a matter of feeding in `.SubModel` in the second case. So, if a user were to see *both* they might reasonably assume that there was a reason for us adding the second -- but there wasn't, actually! It was just a convenience to save the user 9 characters, since we couldn't really use the calibrator anyway, really, right?

Reply:
My opinion might change in a situation where the calibrator was *somehow* being used in a way that was not completely trivial -- e.g., you might imagine a calibrated model in a logistic model getting model parameters that are somehow "adjusted" by the behavior of the calibrator. But that certainly is not the case now, so it should not be here.